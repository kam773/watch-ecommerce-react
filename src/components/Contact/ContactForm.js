import React from "react";
import styled from "styled-components";
import { Input, Form, FormGroup, Row, Col } from "reactstrap";

export default function ContactForm() {
  return (
    <FormWrapper>
      <Form action="mail.php" id="contact-form" method="POST">
        <Row>
          <Col md="6">
            <FormGroup className="md-form my-3">
              <Input
                type="text"
                id="name"
                name="name"
                placeholder="Your name"
                className="form-control"
              />
            </FormGroup>
          </Col>
          <Col md="6">
            <FormGroup className="md-form my-3">
              <Input
                type="text"
                id="email"
                name="email"
                placeholder="Email"
                className="form-control"
              />
            </FormGroup>
          </Col>
        </Row>
        <Row>
          <Col md="12">
            <FormGroup className="md-form my-3">
              <Input
                type="text"
                id="subject"
                name="subject"
                placeholder="Subject"
                className="form-control"
              />
            </FormGroup>
          </Col>
        </Row>
        <Row>
          <Col md="12">
            <FormGroup className="md-form my-3">
              <Input
                type="textarea"
                id="message"
                name="message"
                placeholder="Message"
                className="form-control md-textarea"
              />
            </FormGroup>
          </Col>
        </Row>
      </Form>
    </FormWrapper>
  );
}

const FormWrapper = styled.div`
  textarea {
    resize: none;
    height: 6rem;
    border: 1px solid #555;
    font-size: 1.2rem;
  }
  input {
    height: 3rem;
    border: 1px solid #555;
    font-size: 1.2rem;
  }
  #contact-form {
    width: 100%;
  }
`;
