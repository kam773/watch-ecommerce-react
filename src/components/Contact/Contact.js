import React from "react";
import styled from "styled-components";
import ContactForm from "./ContactForm";
import { Container, Row, Col } from "reactstrap";

export default function Contact() {
  return (
    <Container fluid>
      <h1 className="text-center text-black text-capitalize my-3">
        Contact us
      </h1>
      <FormWrapper>
        <Row>
          <Col md="4">
            <div className="address-box my-3">
              <div className="address my-3">
                <h5>Address</h5>
                <p>
                  The company name,
                  <span>Lorem ipsum dolor,</span>
                  Glasglow Dr 40 Fe 72.
                </p>
              </div>
              <div className="address my-3">
                <h5>Address1</h5>
                <p>
                  Tel:1115550001,
                  <span>Fax:190-4509-494</span>
                  Email:{" "}
                  <a href="mailto:example@email.com">contact@example.com</a>
                </p>
              </div>
            </div>
          </Col>
          <Col md="8">
            <Row>
              <Col md="12" className="mb-md-0 mb-5">
                <ContactForm />
              </Col>
            </Row>
          </Col>
        </Row>
      </FormWrapper>
    </Container>
  );
}
const FormWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex: 0 0 30%;
  .address-box {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }
  .address {
    font-size: 1.2rem;
  }
  .btn {
    padding: 1rem 2rem;
    border: 2px solid #555;
    background: transparent;
    color: #555;
    font-size: 1.2rem;
    font-weight: 600;
    transition: all 0.3s ease-in-out;
  }
  .btn:hover {
    background: #555;
    color: #fff !important;
  }
`;
