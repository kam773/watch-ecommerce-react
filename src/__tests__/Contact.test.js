import React from 'react';
import Contact from '../components/Contact/Contact';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe("Contact", () => {
    it("renders without crashes", () => {
        const tree = renderer.create(<Contact />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("has h1 tag with class", () => {
        const wrapper = shallow(<Contact/>);
        expect(wrapper.find('h1').hasClass('text-center')).toBeTruthy();
    });
    it('has h1 tag with class', () => {
        const wrapper = mount(<Contact/>);
        const post = wrapper.find('h1');
        expect(['text-center', 'text-black', 'text-capitalize', 'my-3'].every(c => post.hasClass(c))).toEqual(true);
    });
    it('has p tag', () => {
        const wrapper = mount(<Contact/>);
        const post = wrapper.find('p');
        expect(post.exists()).toEqual(true);
    });
    it('has a form element', () => {
        const wrapper = mount(<Contact/>);
        const post = wrapper.find('form');
        expect(post.exists()).toEqual(true);
    });
    it('has input element', () => {
        const wrapper = mount(<Contact/>);
        const post = wrapper.find('input');
        expect(post.exists()).toEqual(true);
    });
    it('has anchor element', () => {
        const wrapper = mount(<Contact/>);
        const post = wrapper.find('a');
        expect(post.exists()).toEqual(true);
    });
});