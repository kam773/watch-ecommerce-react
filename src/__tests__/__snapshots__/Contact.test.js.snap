// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders without crashes 1`] = `
<section
  className="container-fluid"
>
  <h1
    className="text-center text-black text-capitalize my-3"
  >
    Contact us
  </h1>
  <div
    className="sc-bdVaJa iZvBxM"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <div
          className="address-box my-3"
        >
          <div
            className="address my-3"
          >
            <h5>
              Address
            </h5>
            <p>
              The company name,
              <span>
                Lorem ipsum dolor,
              </span>
              Glasglow Dr 40 Fe 72.
            </p>
          </div>
          <div
            className="address my-3"
          >
            <h5>
              Address1
            </h5>
            <p>
              Tel:1115550001,
              <span>
                Fax:190-4509-494
              </span>
              Email: 
              <a
                href="mailto:example@email.com"
              >
                contact@example.com
              </a>
            </p>
          </div>
        </div>
      </div>
      <section
        className="col-md-8"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 mb-md-0 mb-5"
          >
            <form
              action="mail.php"
              id="contact-form"
              method="POST"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <div
                    className="md-form my-3"
                  >
                    <input
                      className="form-control"
                      id="name"
                      name="name"
                      placeholder="Your name"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <div
                    className="md-form my-3"
                  >
                    <input
                      className="form-control"
                      id="email"
                      name="email"
                      placeholder="Email"
                      type="text"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-12"
                >
                  <div
                    className="md-form my-3"
                  >
                    <input
                      className="form-control"
                      id="subject"
                      name="subject"
                      placeholder="Subject"
                      type="text"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-12"
                >
                  <div
                    className="md-form my-3"
                  >
                    <textarea
                      className="form-control md-textarea"
                      id="message"
                      name="message"
                      placeholder="Message"
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </form>
            <div
              className="my-3"
            >
              <a
                className="btn"
              >
                Send
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</section>
`;
